src/ds5/ds5-options.cpp

```diff
@@ -732,12 +734,62 @@ namespace librealsense
         if (res.empty())
             throw invalid_value_exception("auto_exposure_limit_option::query result is empty!");
 
-        return *(reinterpret_cast<uint32_t*>(res.data() + 4));
+        return static_cast<float>(*(reinterpret_cast<uint32_t*>(res.data() + 4)));
     }
 
     option_range auto_gain_limit_option::get_range() const
     {
         return *_range;
     }
 
+    librealsense::thermal_compensation::thermal_compensation(
+        std::shared_ptr<ds5_thermal_monitor> monitor,
+        std::shared_ptr<option> toggle) :
+        _thermal_monitor(monitor),
+        _thermal_toggle(toggle)
+    {
+        try {
+            auto on = static_cast<bool>(_thermal_toggle->query());
+            _thermal_monitor->update(on);
+        }
+        catch (...)
+        {
+            LOG_WARNING("Initial Thermal Compensation control state could not be established");
+        }
+    }
+
+    float librealsense::thermal_compensation::query(void) const
+    {
+        auto val = _thermal_toggle->query();
+        _thermal_monitor->update(static_cast<bool>(val));
+        _recording_function(*this);
+        return val;
+    }
+
+    void librealsense::thermal_compensation::set(float value)
+    {
+        if (value < 0)
+            throw invalid_value_exception("Invalid input for thermal compensation toggle: " + std::to_string(value));
+
+        _thermal_toggle->set(value);
+        _thermal_monitor->update(static_cast<bool>(value));
+        _recording_function(*this);
+    }
+
+    const char* librealsense::thermal_compensation::get_description() const
+    {
+        return "Toggle thermal compensation adjustments mechanism";
+    }
+
+    const char* librealsense::thermal_compensation::get_value_description(float value) const
+    {
+        if (value == 0)
+        {
+            return "Disable thermal compensation";
```

Comment:
Shouldn't this have the opposite action?
If it's off, it should either say `Enable thermal compensation` or `Thermal compensation is disabled`.

Reply:
Thanks, swapped